### DISCUSSION TITLE
use ex.Engine without canvas (disable all drawing processes)

### DISCUSSION THREAD

**Discussion by plemnoooooo** [excaliburjs/Excalibur#3033] [Q&A] [answered]
What is the recommended way to disable ex.Engine's drawing processes and canvas? (just using Excalibur for physics, movement, etc.)

**Comment by eonarheim** [comment, marked as answer]
@plemnoooooo Good question! A more official feature is planned to support a headless mode, but unfortunately it's a fair amount of work to untangle the drawing/browser dependencies from the core loop.

A few folks have had luck providing a shim for the draw to effectively disable it https://github.com/excaliburjs/Excalibur/discussions/2759#discussioncomment-7745845

Other folks have taken the approach re-implementing physics/movement on the server for the purposes of multiplayer, they also wind up running at a slower tick rate as well.
